Library/ViewModels/DashboardViewModel.swift

```diff
@@ -87,34 +102,69 @@ public protocol DashboardViewModelType {
 public final class DashboardViewModel: DashboardViewModelInputs, DashboardViewModelOutputs,
   DashboardViewModelType {
 
-    public init() {
+  //Project selected by user in the drawer or by push for the project message thread
+  private let selectedProjectProperty: SignalProducer<Param?, NoError>
```

Comment:
These private consts could be instantiated in the init and not live on the view model.